### ISSUE TITLE
OpenTimelineIO resolver plugin PoC

### ISSUE THREAD

**Issue by foundrytom** [TheFoundryVisionmongers/OpenAssetIO#215]
## What

Produce a minimal prototype [OpenTimelineIO plugin](https://github.com/OpenTimelineIO/otio-plugin-template) that allows `MediaReferences` to be resolved through OpenAssetIO.

## Why

Enabling OpenAssetIO resolution for OpenTimelineIO media references provides a path of minimal resistance to demonstrate how the concepts of OpenAssetIO allows users to define portable or asset-centric EDLs. It also provides an additional layer of validation that the currently proposed API surface area is fit for purpose.

## Acceptance Criteria

- An OpenTimelineIO plugin (`otio-openassetio`) in its own repository that provides an [OTIO Media Linker](https://opentimelineio.readthedocs.io/en/latest/tutorials/write-a-media-linker.html) that resolves any Entity References discovered in Media References using the `ExternalReference` schema.
- The plugin should follow all OTIO best practices.
- The manager identifier to use for resolution should be identified using the `OPENASSETIO_OTIO_MANAGER_IDENTIFIER` environment variable.
- A new simple, "mapping" `ManagerPlugin` within the `OpenAssetIO` project that can be used to facilitate testing and simple workflow experiments:
  - `BasicAssetLibrary` (it can only resolve existing assets in a pre-defined library, no registration or updates at this time)
  - Entity References require the `bal:///` prefix. (`b`asic `a`sset `l`ibrary)
  - Configured with a single `.json` file that defines the primary string for any number of known entity references:
     ```
     { "entities" : { "<name>" : { "versions": [ { "primary-string": "some/path" }, ... ], ... } }
     ```
     This format allows for future expansion to add `attributes`, `versioning`, basic relationships etc...
- E2E tests that demonstrate OTIO python API access to the resolved paths in a suitably configured `.otio` file, using only OTIO API calls.

## Out of scope (future work)
- Available frame ranges.
- Configuring the manager, its settings or resolution context local data from OTIO metadata.

**Comment by foundrytom** [comment]
@feltech had a much better idea for a name `ReadOnlyManager`.

**Comment by foundrytom** [comment]
> @feltech had a much better idea for a name `ReadOnlyManager`.

Based on feedback from the recent working group, we should rethink this a little. We need a simple mechanism to demonstrate some canonical, and some illustrative workflows for "the basics of how the API work". This really requires mild state persistence, so that basic registration can be demonstrated.

Updated the proposal to `BasicAssetLibrary`, so that we can add other core API functionality over time, with the design that this would always be **constrained to the "librarian" model** - ie: a non-path-managing manager. SAM would be a full path-managing manager (#115 will change this to a 'future entity capable' or similar).

**Comment by foundrytom** [comment]
https://academysoftwarefdn.slack.com/archives/CMQ9J4BQC/p1646932031108329?thread_ts=1639076779.387100&cid=CMQ9J4BQC

**Comment by foundrytom** [comment]
Initial PoC is working, will updated with new workflows under a separate issue once #269 is complete.
